exif.cc

```diff
@@ -977,24 +977,9 @@ exif_waypt_from_exif_app(ExifApp* app)
   }
 
   if (opt_filename) {
-    char* c;
-    char* str = xstrdup(fin->name);
-
-    char* cx = str;
-    if ((c = strrchr(cx, ':'))) {
-      cx = c + 1;
-    }
-    if ((c = strrchr(cx, '\\'))) {
-      cx = c + 1;
-    }
-    if ((c = strrchr(cx, '/'))) {
-      cx = c + 1;
-    }
-    if (((c = strchr(cx, '.'))) && (c != cx)) {
-      *c = '\0';
-    }
-    wpt->shortname = cx;
-    xfree(str);
+    QFileInfo fi(fin_->name);
+    // No directory, no extension.
+    wpt->shortname = fi.baseName();
```

Comment:
Nice.   It's not exact, I believe the old code would translate ".bashrc" to ".bashrc",  with baseName() it goes to an empty non-null string.

Reply:
To be clear I am not suggesting we make the old and new code exactly equivalent.  I think baseName() is fine.